version: "3.3"
services:
  traefik:
    image: traefik:v2.5.6
    container_name: traefik
    restart: unless-stopped
    command: 
      - "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--api.dashboard=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.web_public.address=:22715"
      - "--entrypoints.web_public_secure.address=:38516"
      - "--certificatesResolvers.lets_encrypt.acme.httpChallenge=true"
      - "--certificatesResolvers.lets_encrypt.acme.email=oskar@oskarrosen.com"
      - "--certificatesResolvers.lets_encrypt.acme.storage=acme.json"
      - "--certificatesResolvers.lets_encrypt.acme.httpChallenge.entryPoint=web_public"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    networks:
      - default
      - traefik_internal
    ports:
      - "80:80"
      - "8080:8080"
      - "22715:22715"
      - "38516:38516"
 
networks:
  traefik_internal:
   external: true
